dataplane/src/pipeline/mod.rs

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+//! # Pipeline Building Blocks
+//!
+//! This module provides the building blocks for constructing pipelines of network functions.
+//! There are two main methods provided for linking network functions together in sequence:
+//!
+//! - `StaticChain`: A trait for statically chaining network functions together.
+//! - `DynPipeline`: A pipeline that can be dynamically constructed at runtime.
+//!
+//! ## Network Functions
+//!
+//! A network function is any object that implements the [`NetworkFunction`] trait.
+//! You can look at the [`sample_nfs`] module for some examples of simple network functions.
```

Comment:
The word "object" is somewhat overloaded in the context of rust in that many folks use it to refer to anything which is `dyn` safe.  I would recommend swapping "object" for `struct` or instead say "A network function is anything that ..."

Reply:
Done.  I reworded to "anything"